modules/development/src/main/java/unnamed/mmo/server/dev/blocks/ore/OreCreatorTool.java

```diff
@@ -0,0 +1,109 @@
+        return ItemStack.builder(Material.DIAMOND)
+                .displayName(ComponentUtil.fromStringSafe("Ore Creator"))
+                .lore(
+                        ComponentUtil.fromStringSafe("<white>Selection: None"),
+                        ComponentUtil.fromStringSafe("<white>Shift+RMB: Select ore"),
+                        ComponentUtil.fromStringSafe("<white>RMB: Assign selected ore")
+                )
+                .build();
+    }
+
+    @Override
+    public @NotNull ItemStack rightClicked(@NotNull Player player, @NotNull ItemStack itemStack, @Nullable Point targetBlock, @Nullable Entity targetEntity) {
+        return player.isSneaking() ?
+                openOreSelector(player, itemStack) :
+                assignOre(player, itemStack, targetBlock);
+    }
+
+    private ItemStack assignOre(Player player, ItemStack itemStack, Point targetBlock) {
+        if (targetBlock == null)
+            return itemStack;
+
+        final Ore ore = Ore.fromNamespaceId(itemStack.getTag(SELECTION_TAG));
+        if (ore == null) {
+            player.sendMessage("You must select an ore before placing it.");
+            return itemStack;
+        }
+
+        final Instance instance = player.getInstance();
+        instance.setBlock(targetBlock, ore.asBlock());
+        player.sendMessage("Set " + targetBlock + " to " + ore.name() + "!");
+        return itemStack;
+    }
+
+    private ItemStack openOreSelector(Player player, ItemStack itemStack) {
+        final Inventory oreSelector = new Inventory(InventoryType.CHEST_2_ROW, "Ore Selector");
+        oreSelector.addInventoryCondition(this::selectOre);
+        Ore.REGISTRY.values().stream()
+                .sorted(Comparator.comparing(Ore::name))
+                .forEach(ore -> {
+                    //todo is there a better Block->Material conversion
+                    Material oreMaterial = Material.fromNamespaceId(ore.oreBlock().namespace());
+                    if (oreMaterial == null) oreMaterial = Material.STONE;
+                    final ItemStack oreItem = ItemStack.builder(oreMaterial)
+                            .displayName(ComponentUtil.fromStringSafe(ore.name()))
+                            .build()
+                            .withTag(SELECTION_TAG, ore.name());
+                    oreSelector.addItemStack(oreItem);
+                });
+        player.openInventory(oreSelector);
+        return itemStack;
+    }
+
+    private void selectOre(@NotNull Player player, int slot, @NotNull ClickType clickType, @NotNull InventoryConditionResult result) {
+        result.setCancel(true);
+        if (clickType != ClickType.LEFT_CLICK) return;
+
+        final Inventory inventory = player.getOpenInventory();
+        if (inventory == null) return;
+        final ItemStack clickedItem = inventory.getItemStack(slot);
+        if (clickedItem.isAir()) return;
+
+        //todo is it safe to just use main hand here? is rightCLick triggered on off hand? probably!
```

Comment:
You can right click with off hand, so this isn't safe

Reply:
I ignored the off hand events for now. The left click ones using hand animation only do main hand for now. If we support off hand for that then this should be changed (also include hand in left/rightClicked methods I guess)